tools/webidl_binder.py

```diff
@@ -485,20 +480,28 @@ def render_function(class_name, func_name, sigs, return_type, non_pointer, copy,
   for i in range(min_args, max_args+1):
     raw = sigs.get(i)
     if raw is None: continue
-    sig = list(map(full_typename, raw))
-    if array_attribute:
-      sig = [x.replace('[]', '') for x in sig] # for arrays, ignore that this is an array - our get/set methods operate on the elements
 
-    c_arg_types = list(map(type_to_c, sig))
+    if sequence_attribute:
+      # For array attributes, the get/set methods operate on the
+      # elements, not the array as a whole.
+      arg_types = [arg.type.inner if arg.type.isSequence() else arg.type
+                   for arg in raw]
+    else:
+      arg_types = [arg.type for arg in raw]
+
+    const_qualifiers = ['const ' if arg.getExtendedAttribute('Const') else ''
+                        for arg in raw]
```

Comment:
the const code seems like a new addition - I don't see any previous handling of const that it replaces? Is that a bugfix?